content/en/docs/concepts/architecture/unknown-version-interoperability-proxy.md

```diff
@@ -0,0 +1,79 @@
+
+This mechanism is called the _Unknown Version Interoperability Proxy_.
+## Enabling the Unknown Version Interoperability Proxy
+The Unknown Version Interoperability Proxy feature is controlled by a
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+and is not enabled by default. You can activate Unknown Version Interoperability Proxy by enabling
+the feature gate named `UnknownVersionInteroperabilityProxy` when you start the
+{{< glossary_tooltip text="API Server" term_id="kube-apiserver" >}}:
+
+```shell
+kube-apiserver \
+--feature-gates=UnknownVersionInteroperabilityProxy=true \
+# required flag for this feature
+--peer-ca-file=`ca bundle to verify serving certs of kube-apiserver`
+# optional flags for this feature
+--peer-advertise-ip=`ip of this kube-apiserver that should be used by peers to proxy requests`
+--peer-advertise-port=`port of this kube-apiserver that should be used by peers to proxy requests`
+# …and other flags as usual
+```
+
+## Interoperability proxying
+
+When you enable interoperability proxying, the [aggregation layer](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) 
+loads a special filter that identifies API groups / versions / resources that the local server doesn't recognise,
+and tries to proxy those requests to a peer API server that is capable of handling the request.
+
+When a kube-apiserver receives a resource request, it first checks which API servers can serve the
+requested resource. The check happens using the internal `StorageVersion` API.
+If the resource is known to the local API server, the request is handled locally. If there is no 
+internal `StorageVersion` object found for the requested resource and the configured APIService
+specifies proxying to an extension API server, that proxying happens following the usual
+[flow](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/) for
+extension APIs.
+
+Once a valid internal `StorageVersion` object is found for the requested resource, 
+the API server trying to handle the request (the _handling API server_) fetches the 
+peer API servers that do serve the relevant API group / version / resources using the information 
+in the fetched `StorageVersion` object. The _handling API server_ then proxies the request
+to one of the matching peer kube-apiservers that are aware of the requested resource.
+If there is no peer known for that API group / version / resource, the handling API server passes the 
+request to its own handler chain which should eventually return a 404("Not Found") response.
+If the handling API server has identified and selected a peer API server, but that peer fails to respond
+(for reasones such as network connectivity, or a data race between the request being received and
+a controller registering the peer's info into an internal `ServerStorageVersion` object), then the
+handling API server responds with a 503 (“Service Unavailable”) error.
```

Comment:
Thread an example into this? It might be hard for someone unfamiliar with the idea to unpack all of this.

Reply:
Added different examples for different scenarios PTAL